test/jdk/java/nio/channels/SocketChannel/OpenLeak.java

```diff
@@ -26,27 +26,77 @@
  * @summary SocketChannel.open(SocketAddress) leaks file descriptor if
  *     connection cannot be established
  * @build OpenLeak
- * @run main/othervm -Djava.security.manager=allow OpenLeak
+ * @run junit/othervm OpenLeak
  */
 
+import java.net.ConnectException;
 import java.net.InetAddress;
 import java.net.InetSocketAddress;
+import java.net.SocketAddress;
 import java.nio.channels.SocketChannel;
+import java.nio.channels.UnresolvedAddressException;
+
+import org.junit.jupiter.api.Assumptions;
+import org.junit.jupiter.api.Test;
 
 public class OpenLeak {
 
-    public static void main(String[] args) throws Exception {
-        InetAddress lh = InetAddress.getLocalHost();
-        InetSocketAddress isa = new InetSocketAddress(lh, 12345);
+    private static final int MAX_LOOP = 250000;
```

Comment:
Overall changes looks good to me but do you need to loop  25000 times ?.

Reply:
Actually 250000 times ;-)

Yes - without the close() the test starts failling for me on my machine at around 247500 iterations. So if we really want to test that the issue the original test was trying to verify is fixed we need that, I believe.